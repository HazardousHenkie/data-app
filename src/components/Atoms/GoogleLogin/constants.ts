import LoginState from './types'

enum ActionTypes {
    LOGIN_REQUEST = 'components/GoogleLogin/LOGIN_REQUEST',
    LOGIN_SUCCESS = 'components/GoogleLogin/LOGIN_SUCCESS',
    LOGIN_ERROR = 'components/GoogleLogin/LOGIN_ERROR'
}

export const initialUserState: LoginState['user'] = {
    id: 0,
    name: ''
}

export const initialLoginState: LoginState = {
    loggedIn: false,
    error: false,
    loading: false,
    user: initialUserState
}

export default ActionTypes
