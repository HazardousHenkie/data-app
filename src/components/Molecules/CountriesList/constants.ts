enum ActionTypes {
    SET_SELECTED_COUNTRY = 'components/CountriesList/SET_SELECTED_COUNTRY',
    GET_COUNTRIES_DATA = 'components/CountriesList/GET_COUNTRIES_DATA',
    GET_COUNTRIES_DATA_SUCCESS = 'components/CountriesList/GET_COUNTRIES_DATA_SUCCESS',
    GET_COUNTRIES_DATA_ERROR = 'components/CountriesList/GET_COUNTRIES_DATA_ERROR'
}

export default ActionTypes
