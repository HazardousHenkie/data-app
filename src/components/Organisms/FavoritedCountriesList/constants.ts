enum ActionTypes {
    GET_FAVORITED_COUNTRIES = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES',
    GET_FAVORITED_COUNTRIES_SUCCESS = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES_SUCCESS',
    GET_FAVORITED_COUNTRIES_ERROR = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES_ERROR'
}

export default ActionTypes
