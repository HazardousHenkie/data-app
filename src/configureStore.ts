import { applyMiddleware, createStore, compose } from 'redux'
import { routerMiddleware } from 'connected-react-router'
import { createInjectorsEnhancer, forceReducerReload } from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'
import { History } from 'history'
import { createLogger } from 'redux-logger'

import { InjectedStore, ApplicationRootState } from 'types'
import createReducer from './reducers'

export default function configureStore(
    initialState: ApplicationRootState | object,
    history: History
): InjectedStore {
    const sagaMiddleware = createSagaMiddleware()
    const { run: runSaga } = sagaMiddleware
    const middlewares = [sagaMiddleware, routerMiddleware(history)]

    const enhancers = [
        applyMiddleware(...middlewares),
        createInjectorsEnhancer({
            createReducer,
            runSaga
        })
    ]

    if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
        const logger = createLogger()
        middlewares.push(logger)
    }

    const store = createStore(
        createReducer(),
        initialState as object,
        compose(...enhancers)
    ) as InjectedStore

    // eslint-disable-next-line
    store.runSaga = sagaMiddleware.run
    store.injectedReducers = {}
    store.injectedSagas = {}

    if (module.hot) {
        module.hot.accept('./reducers', () => {
            forceReducerReload(store)
        })
    }

    return store
}
