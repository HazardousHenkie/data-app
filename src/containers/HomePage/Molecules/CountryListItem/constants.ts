import CountryState from './types'

enum ActionTypes {
    SET_SELECTED_COUNTRY = 'components/CountriesList/SET_SELECTED_COUNTRY'
}

export const CountryItem: CountryState = {
    country: {
        alpha2Code: '',
        name: '',
        nativeName: '',
        capital: '',
        region: '',
        subregion: '',
        flag: '',
        currency: '',
        population: 0,
        latlng: [0, 0],
        currencies: [{ string: '' }],
        languages: [{ string: '' }],
        translations: { string: '' }
    }
}

export default ActionTypes
