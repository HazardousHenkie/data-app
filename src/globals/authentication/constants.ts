import AuthenticationState, { User } from './types'

export const initialUserState: User = {
    id: '',
    name: ''
}

const initialAuthenticationState: AuthenticationState = {
    loggedIn: false,
    error: false,
    loading: false,
    user: initialUserState
}

export default initialAuthenticationState
