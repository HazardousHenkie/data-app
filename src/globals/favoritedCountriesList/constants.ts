import { query } from 'faunadb'
import FavoritedCountriesState from './types'

enum ActionTypes {
    GET_FAVORITED_COUNTRIES = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES',
    SET_FAVORITED_COUNTRIES = 'components/FavoritedCountriesList/SET_FAVORITED_COUNTRIES',
    GET_FAVORITED_COUNTRIES_ERROR = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES_ERROR'
}

export const initialFavoritedCountriesState: FavoritedCountriesState = {
    error: false,
    loading: false,
    countries: [
        {
            ref: {
                '@ref': {
                    collection: query.Collection('country_user'),
                    id: ''
                }
            },
            ts: 0,
            data: {
                userId: '',
                countryId: '',
                updatedAt: 0,
                createdAt: 0
            }
        }
    ]
}

export default ActionTypes
