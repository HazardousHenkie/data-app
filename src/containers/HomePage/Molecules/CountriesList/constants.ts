import { CountryItem } from 'containers/HomePage/Molecules/CountryListItem/constants'
import CountriesListState from './types'

enum ActionTypes {
    GET_COUNTRIES_DATA = 'components/CountriesList/GET_COUNTRIES_DATA',
    GET_COUNTRIES_DATA_SUCCESS = 'components/CountriesList/GET_COUNTRIES_DATA_SUCCESS',
    GET_COUNTRIES_DATA_ERROR = 'components/CountriesList/GET_COUNTRIES_DATA_ERROR'
}

export const initialCountriesHeaderState: CountriesListState = {
    error: false,
    loading: false,
    data: [CountryItem.country]
}

export default ActionTypes
