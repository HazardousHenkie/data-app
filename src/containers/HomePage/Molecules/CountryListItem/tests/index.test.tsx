import React from 'react'
import { render } from '@testing-library/react'

import { Provider } from 'react-redux'
import configureStore from 'configureStore'
import history from 'utils/history'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

import ListItem from '../index'
import { CountryItem } from '../constants'

describe('<ListItem />', () => {
    it('should render like snapshot', () => {
        const initialState = {}
        const store = configureStore(initialState, history)

        const {
            container: { firstChild }
        } = render(
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            <ListItem
                                listCountry={CountryItem.country}
                                setOpen={() => {}}
                            />
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )

        expect(firstChild).toMatchSnapshot()
    })
})
