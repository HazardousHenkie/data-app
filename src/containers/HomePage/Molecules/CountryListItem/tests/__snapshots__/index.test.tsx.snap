// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render like snapshot 1`] = `
<div
  aria-disabled="false"
  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  role="button"
  tabindex="0"
>
  <div
    class="MuiListItemIcon-root"
  >
    <img
      alt=""
      height="20"
      src=""
      width="30"
    />
  </div>
  <div
    class="MuiListItemText-root"
  />
  <span
    class="MuiTouchRipple-root"
  />
</div>
`;
